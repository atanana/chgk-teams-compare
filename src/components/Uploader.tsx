import * as React from 'react';

const Uploader = () => (
    <div>
        <div className="file">
            <label className="file-label">
                <input className="file-input" type="file" name="resume"/>
                <span className="file-cta">
                    <span className="file-icon">
                        <i className="fa fa-upload"/>
                    </span>
                    <span className="file-label">
                        Выбрать файл…
                    </span>
                </span>
            </label>
        </div>
    </div>
);

export default Uploader;