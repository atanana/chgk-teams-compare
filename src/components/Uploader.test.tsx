import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Uploader from './Uploader';
import Mock = jest.Mock;

const onFileSelectedMock: Mock<(files: FileList | null) => void> = jest.fn();

it('renders without crashing', () => {
    const component = renderer.create(<Uploader onFileSelected={onFileSelectedMock}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('calls callback on file select', () => {
    const uploader = shallow(<Uploader onFileSelected={onFileSelectedMock}/>);
    const input = uploader.find('input');
    const files: Mock<FileList> = jest.fn();
    input.simulate('change', { target: { files } });

    expect(onFileSelectedMock.mock.calls.length).toBe(1);
    expect(onFileSelectedMock.mock.calls[0][0]).toBe(files);
});