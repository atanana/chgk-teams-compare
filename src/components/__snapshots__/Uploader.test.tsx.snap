// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    className="file"
  >
    <label
      className="file-label"
    >
      <input
        className="file-input"
        onChange={[Function]}
        type="file"
      />
      <span
        className="file-cta"
      >
        <span
          className="file-icon"
        >
          <i
            className="fa fa-upload"
          />
        </span>
        <span
          className="file-label"
        >
          Выбрать файл…
        </span>
      </span>
    </label>
  </div>
</div>
`;
